### PR TITLE
Bids 2245/split notification process

### PR DESCRIPTION
**Pull request by peterbitfly**
Split the notification process into two binaries, one responsible for queuing new notifications and one for sending notifications out. Improves marking webhook notifications as sent and harmonizes logging message in the notification module.

### <samp>🤖 Generated by Copilot at a8e1d31</samp>

This pull request refactors the notification service of the explorer into two new binaries: `notification-sender` and `notification-collector`. It also improves the logging, error handling, and database connection management of the notification system. It updates the `Makefile` and the `cmd/explorer/main.go` file to support the new binaries.

### CODE REVIEW

File: cmd/notification-sender/main.go [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+	}
+
+	wg.Wait()
+	if utils.Config.TieredCacheProvider == "bigtable" && len(utils.Config.RedisCacheEndpoint) == 0 {
+		cache.MustInitTieredCacheBigtable(db.BigtableClient.GetClient(), fmt.Sprintf("%d", utils.Config.Chain.Config.DepositChainID))
+		logrus.Infof("tiered Cache initialized, latest finalized epoch: %v", services.LatestFinalizedEpoch())
+	}
+
+	if utils.Config.TieredCacheProvider != "bigtable" && utils.Config.TieredCacheProvider != "redis" {
+		logrus.Fatalf("no cache provider set, pease set TierdCacheProvider (example redis, bigtable)")
```
Comment (recy21) [inline cmd/notification-sender/main.go:124]: typo in please (missing L)

File: cmd/notification-sender/main.go [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+			cfg.Frontend.WriterDatabase.Name}
+		frontendDBStr := strings.Join(frontendDBInfo, "-")
+		if DBStr != frontendDBStr {
+			go metrics.MonitorDB(db.FrontendWriterDB)
+		}
+	}
+
+	logrus.Infof("database connection established")
+
+	services.InitNotificationSender()
```
Comment (recy21) [inline cmd/notification-sender/main.go:156]: 1:1 same function as in notifications-collector, should be handled in a function. The different line can be called after, or via flag / if

File: services/notifications.go [hunk 3]
```diff
@@ -1011,13 +1012,13 @@ func sendWebhookNotifications(useDB *sqlx.DB) error {
...
 
-			if resp != nil && resp.StatusCode < 400 {
-				_, err := useDB.Exec(`UPDATE notification_queue SET sent = now();`)
-				if err != nil {
-					logger.WithError(err).Errorf("error updating notification_queue table")
-					return
-				}
+			_, err = useDB.Exec(`UPDATE notification_queue SET sent = now() where id = $1`, n.Id)
+			if err != nil {
+				logger.WithError(err).Errorf("error updating notification_queue table")
```
Comment (recy21) [inline services/notifications.go:1017]: Should use `utils.LogError(err,"error updating notification_queue table",0)`

**Review by recy21 — CHANGES_REQUESTED** [review]
As `Config.Notifications.Enabled`is obsolete now, it should be removed from config

File: services/notifications.go (hunk 3 above)
Comment (peterbitfly) [reply services/notifications.go:1017]: we should rework all errors in a separate PR and not in a single location in this pr

File: cmd/notification-sender/main.go (hunk 2 above)
Comment (peterbitfly) [reply cmd/notification-sender/main.go:156]: don't understand

File: cmd/notification-sender/main.go (hunk 1 above)
Comment (peterbitfly) [reply cmd/notification-sender/main.go]: fixed

File: cmd/notification-sender/main.go (hunk 2 above)
Comment (recy21) [reply cmd/notification-sender/main.go:156]: `cmd/notification-sender/main.go > func main()`
and 
`cmd/notification-collector/main.go > func main()`
differ only by 1 line.

So I suggest to have only 1 function for this, where the 1 different line is called outside or called via an flag.

File: types/config.go [hunk 4]
```diff
@@ -169,7 +169,6 @@ type Config struct {
 		Pprof   bool   `yaml:"pprof" envconfig:"METRICS_PPROF"`
 	} `yaml:"metrics"`
 	Notifications struct {
-		Enabled                                       bool   `yaml:"enabled" envconfig:"NOTIFICATIONS_ENABLED"`
 		Sender                                        bool   `yaml:"sender" envconfig:"NOTIFICATIONS_SENDER"`
```
Comment (recy21) [inline types/config.go:172]: Please remove Sender too, isn't used anymore

**Review by recy21 — CHANGES_REQUESTED** [review]

**Review by recy21 — APPROVED** [review]

**Pull request merged**